### PR TITLE
Release notes 2022.3

### PR DESCRIPTION
**Pull request by viniarck**
Initial draft for release notes 2022.3

There are TODOs that need to be done and to add more information. 

As of now, I updated the document based on the current status of issues/epics that have landed.

### CODE REVIEW

**Comment by viniarck** [comment]
I'll set this PR as ready for review, but ready for review means that it can be reviewed for the parts that don't have any TODO mentioned. Once [these issues are closed](https://github.com/search?p=1&q=org%3Akytos-ng+org%3Aamlight+label%3A2022.3+is%3Aopen&type=Issues), then the statistics need to be updated and update other details.

File: releases/2022.3/2022.3.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+In addition, `mef_eline` provides service level prioritization to allow EVCs to be handled based on their descending `service_level` value during convergence and consistency checks.
+
+### 2. Specify how INT will be integrated and supported via NoviFlow experimenter actions
+
+A new `telemetry` has been specified on blueprint [EP031](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP031.rst), which will be responsible for enabling INT on EVCs. NoviFlow switches will be supported via experimenter actions that been implemented by the `noviflow` NApp, and the functionality will be optionally enabled when an EVC is being created. The `telemetry` NApp is expected to be available in the `2023.1` release.
+
+### 3. `maintenance` stabilization v1 API and integrate with MongoDB 
+
+**TODO** fill out what will be feasible to be delivered, ask David to update this paragraph
```
Comment (viniarck) [inline releases/2022.3/2022.3.md:46]: FYI @Ktmi, if you could help out to summarize and update this document later that'd be great, thanks.

File: releases/2022.3/2022.3.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+### 4. Support NoviFlow switches on end-to-end tests
+
+NoviFlow switches are now supported on [kytos-end-to-end-tests](https://github.com/amlight/kytos-end-to-end-tests/) as a target on pre-defined [testing topologies](https://github.com/amlight/kytos-end-to-end-tests/blob/feature/noviswitch/tests/helpers.py). Currently, NoviFlow switches are managed via SSH using `paramiko` and this integration was designed to be seamless with mininet, NoviFlow switches are handled as a mininet `Switch` so you can reuse or write new end-to-end tests.
```
Comment (viniarck) [inline releases/2022.3/2022.3.md:50]: @italovalcy please feel free to update any other details that might be relevant here, I tried to summarize based on the latest updates. Thanks.

File: releases/2022.3/2022.3.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+### 5. end-to-end tests should have fewer false positives and support retries
+
+end-to-end tests were enhanced to avoid performing assertions when network convergence can still be taking place, minimizing false positives asserts that could happen. Also, `pytest-rerunfailures` was introduced to help out renunning tests, that way providing additional evidence facilitating to identify and isolate underlying issues.
+
+### 6. `flow_stats` stabilization v1 API
+
+`flow_stats` has had its API reviewed and augmented consolidating its v1 API. `flow_stats` has simplified its code and responsibilities.
```
Comment (viniarck) [inline releases/2022.3/2022.3.md:58]: @gretelliz feel free to add or complement any other relevant information here if needed. Thanks

File: releases/2022.3/2022.3.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+[kytos/maintenance](https://github.com/kytos-ng/maintenance)   | [kytos-ng/maintenance](https://github.com/kytos-ng/maintenance/releases)
+[amlight/coloring](https://github.com/kytos-ng/coloring)       | [kytos-ng/coloring](https://github.com/kytos-ng/coloring/releases)
+[amlight/sdntrace](https://github.com/kytos-ng/sdntrace)       | [kytos-ng/sdntrace](https://github.com/kytos-ng/sdntrace/releases)
+[amlight/flow_stats](https://github.com/kytos-ng/flow_stats)   | [kytos-ng/flow_stats](https://github.com/kytos-ng/flow_stats/releases)
+[amlight/sdntrace_cp](https://github.com/kytos-ng/sdntrace_cp) | [kytos-ng/sdntrace_cp](https://github.com/kytos-ng/sdntrace_cp/releases)
+
+
+### Switches Compatibility List
+
+**TODO only add NoviFlow info once RC period is confirmed**
```
Comment (viniarck) [inline releases/2022.3/2022.3.md:115]: reminder to nuke this TODO eventually

File: releases/2022.3/2022.3.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+```
+Flask/Werkzeug 2.0.0+ now provide ``async`` support, so NApps can leverage ``asyncio`` and its ecosystem when applicable using the same ``rest`` decorator.
+
+### 10. All Kytos-ng NApps should have at least 85% of unit test coverage
+
+`flow_stats` unit test code coverage has increased to 85%, now all Kytos-ng NApps have at least 85% of unit test coverage. For more information, please visit [Kytos-ng Tests Analytics Dashboard](https://kytos-tests.amlight.net/index.html).
+
+## Kytos-ng and NApps released
+
+**TODO publish the releases when the time comes**
```
Comment (viniarck) [inline releases/2022.3/2022.3.md:91]: reminder to nuke this TODO eventually

File: releases/2022.3/2022.3.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+In addition, `mef_eline` provides service level prioritization to allow EVCs to be handled based on their descending `service_level` value during convergence and consistency checks.
+
+### 2. Specify how INT will be integrated and supported via NoviFlow experimenter actions
+
+A new `telemetry` has been specified on blueprint [EP031](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP031.rst), which will be responsible for enabling INT on EVCs. NoviFlow switches will be supported via experimenter actions that been implemented by the `noviflow` NApp, and the functionality will be optionally enabled when an EVC is being created. The `telemetry` NApp is expected to be available in the `2023.1` release.
```
Comment (italovalcy) [inline releases/2022.3/2022.3.md:42]:
```suggestion
A blueprint has been specified for adding support to `telemetry` in Kytos-ng [EP031](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP031.rst), which will be responsible for enabling INT on EVCs. NoviFlow switches will be supported via experimenter actions that been implemented by the `noviflow` NApp, and the functionality will be optionally enabled when an EVC is being created. The `telemetry` NApp is expected to be available in the `2023.1` release.
```

File: releases/2022.3/2022.3.md (hunk 2 above)
Comment (italovalcy) [inline releases/2022.3/2022.3.md:50]:
```suggestion
NoviFlow switches are now supported on [kytos-end-to-end-tests](https://github.com/kytos-ng/kytos-end-to-end-tests/) as a target on pre-defined [testing topologies](https://github.com/amlight/kytos-end-to-end-tests/blob/feature/noviswitch/tests/helpers.py). Currently, NoviFlow switches are managed via SSH using `paramiko` and this integration was designed to be seamless with Mininet, NoviFlow switches are handled as a Mininet `Switch` so you can reuse or write new end-to-end tests.
```

File: releases/2022.3/2022.3.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+**TODO publish the releases when the time comes**
+
+The following Kytos-ng projects and NApps were released in this version, for further detailed information about what was released and what changed please visit the release page of each project that you are interested in knowing more and look for the changelog file.
+
+
+Project                                                        | Release Web Page
+-------------------------------------------------------------- |------------------------------------------------------------------------
+[kytos](https://github.com/kytos-ng/kytos)                     | [kytos](https://github.com/kytos-ng/kytos/releases)
+[ui](https://github.com/kytos-ng/ui)                           | [ui](https://github.com/kytos-ng/ui/releases)
```
Comment (italovalcy) [inline releases/2022.3/2022.3.md:99]:
```suggestion
[python-openflow](https://github.com/kytos-ng/python-openflow)                           | [python-openflow](https://github.com/kytos-ng/python-openflow/releases)
[kytos-utils](https://github.com/kytos-ng/kytos-utils)                           | [kytos-utils](https://github.com/kytos-ng/kytos-utils/releases)
[ui](https://github.com/kytos-ng/ui)                           | [ui](https://github.com/kytos-ng/ui/releases)
```

File: releases/2022.3/2022.3.md (hunk 1 above)
Comment (Ktmi) [inline releases/2022.3/2022.3.md:46]:
```suggestion
`maintenance` has had its API reviewed and augmented consolidating its v1 API. The `items` field has been removed, and has now been replaced with the `switches`, `links`, and `interfaces` field.  The `status` field has been changed to use a string enum, with the valid possible values being `pending`, `running`, and `finished`. `status` is now protected from being set. Maintenance windows now also include the fields `created_at`, and `updated_at` which contain the datetime of when the field was created and updated. Maintenance windows are now protected from overlapping time intervals, with overlapping disallowed unless the field `force` is set to `True`.

Added persistence to maintenance windows through mongodb. All maintenance windows are stored in the collection `maintenance.windows`. Upon initializing the napp, maintenance will update the status of all windows in the db to progress to the appropriate state depending on the current time compared to there start and end time. Then the napp will deploy all `running` maintenance windows, and schedule all `pending` windows.
```

**Review by italovalcy — APPROVED** [review]
Looks good to me! I added a few updates to the text and updated the statistics.

File: releases/2022.3/2022.3.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+</div>
+
+Kytos-ng[^1] version `2022.3` was released in December
+
+## Objectives
+
+This version had these major objectives:
+
+- `mef_eline` should be able to create EVCs with constrained paths and service level prioritization
+- Specify how INT will be integrated and supported via NoviFlow experimenter actions
```
Comment (italovalcy) [inline releases/2022.3/2022.3.md:22]:
```suggestion
- Add support for INT integration and other NoviFlow experimenter actions
```

**Pull request merged**